### PR TITLE
Buildkite: More comprehensive client sdk job

### PR DESCRIPTION
**Pull request by bkase**
Back-ported circle client sdk changes to buildkite

### CODE REVIEW

File: buildkite/scripts/build-client-sdk.sh [hunk 1]
```diff
@@ -2,5 +2,20 @@
...
 
+echo "--- Make client sdk"
 mkdir -p /tmp/artifacts
 make client_sdk 2>&1 | tee /tmp/artifacts/buildclientsdk.log
+
+echo "--- Get yarn"
+sudo apt install apt-transport-https
+curl -sL https://deb.nodesource.com/setup_14.x | sudo -E bash -
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
```
Comment (lk86) [inline buildkite/scripts/build-client-sdk.sh:13]: these steps seem unreasonable to repeat on each run, can yarn be included in whatever docker image these jobs run from?

Comment (bkase) [reply buildkite/scripts/build-client-sdk.sh:13]: Yes, I suppose you're right. I shall pay the docker rebuild penalty

File: Makefile [hunk 2]
```diff
@@ -180,14 +180,14 @@ macos-setup:
...
+		docker build --file dockerfiles/Dockerfile-toolchain --tag codaprotocol/coda:toolchain-$(GITLONGHASH) . && \
 		docker tag  codaprotocol/coda:toolchain-$(GITLONGHASH) codaprotocol/coda:toolchain-latest && \
 		docker push codaprotocol/coda:toolchain-$(GITLONGHASH) && \
 		docker push codaprotocol/coda:toolchain-latest ;\
-	else \
-		echo "Repo has uncommited changes, commit first to set hash." ;\
-	fi
+	#else \
+		#echo "Repo has uncommited changes, commit first to set hash." ;\
+	#fi
```
Comment (lk86) [inline Makefile:190]: Now that you've presumably committed the docker changes, is commenting out these conditionals still necessary?

Comment (bkase) [reply Makefile:190]: oops, thanks

**Pull request merged**